optuna_integration/mlflow.py

```diff
@@ -0,0 +1,324 @@
+import functools
+import threading
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Sequence
+from typing import Union
```

Comment:
Can we replace them with `from __future__ import annotations`?

```suggestion
from typing import Callable
from typing import Dict
from typing import List
from typing import Optional
from typing import Sequence
from typing import Union
```